orchagent/orch.cpp

```diff
@@ -685,6 +685,46 @@ Executor *Orch::getExecutor(string executorName)
     return NULL;
 }
 
+bool Orch::handleSaiCreateFailure(sai_status_t status)
+{
+    /*
+     * This function aims to provide coarse handling of failures in sairedis create
+     * operation (i.e., notify users by throwing excepions when failures happen).
+     * TODO: 1. Add general handling logic for specific statuses (e.g., SAI_STATUS_ITEM_ALREADY_EXISTS)
+     *       2. Develop fine-grain failure handling mechanisms specific and replace
+     *          this coarse handling in each orch.
+     */
+    SWSS_LOG_THROW("Encountered failure in Create operation, status: %d", status);
```

Comment:
>SWSS_LOG_THROW [](start = 4, length = 14)

In old async mode, [https://github.com/Azure/sonic-swss/blob/master/orchagent/notifications.cpp#L20](https://github.com/Azure/sonic-swss/blob/master/orchagent/notifications.cpp#L20) on_switch_shutdown_request will be called. Do you want to mimic exact message and exit?

Reply:
The message in `on_switch_shutdown_request` is only "syncd stopped". Personally, I do not like this message very much for two reasons: 1. it does not carry much information; 2. It does not precisely describe the situation in the sync mode scenario (In sync mode, syncd hasn't really stopped at this point).

Reply:
Log message is good. Do you still want to mimic the exit behavior?

---
In reply to: [560594351](https://github.com/Azure/sonic-swss/pull/1596#discussion_r560594351) [](ancestors = 560594351)

Reply:
Using exit behavior instead.